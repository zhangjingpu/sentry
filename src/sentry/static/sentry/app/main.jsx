import React from 'react';
import {Router, browserHistory} from 'react-router';
import routes from './routes';

export default class App extends React.Component {
  render() {
    return <Router history={browserHistory}>{routes()}</Router>;
  }
}
