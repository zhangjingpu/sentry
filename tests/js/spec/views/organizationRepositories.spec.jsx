import React from 'react';
import {shallow} from 'enzyme';

import {Client} from 'app/api';
import OrganizationRepositories from 'app/views/settings/organization/repositories/organizationRepositories';

describe('OrganizationRepositories', function() {
  it('renders without providers', function() {
    let wrapper = shallow(
      <OrganizationRepositories
        params={{orgId: 'org-slug'}}
        itemList={[]}
        repoConfig={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with github provider', function() {
    let wrapper = shallow(
      <OrganizationRepositories
        params={{orgId: 'org-slug'}}
        repoConfig={{providers: [TestStubs.GitHubRepositoryProvider()]}}
        itemList={[]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with a repository', function() {
    Client.addMockResponse({
      url: '/organizations/org-slug/repos/',
      body: [TestStubs.Repository()],
    });
    let wrapper = shallow(
      <OrganizationRepositories
        params={{orgId: 'org-slug'}}
        repoConfig={{providers: [TestStubs.GitHubRepositoryProvider()]}}
        itemList={[TestStubs.Repository()]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
