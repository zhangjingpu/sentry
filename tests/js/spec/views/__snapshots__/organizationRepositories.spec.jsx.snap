// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationRepositories renders with a repository 1`] = `
<div>
  <SettingsPageHeading
    action={
      <DropdownLink
        alwaysRenderMenu={true}
        anchorRight={true}
        caret={true}
        className="btn btn-primary btn-sm"
        disabled={false}
        title="Add Repository"
      >
        <MenuItem
          noAnchor={true}
        >
          <AddRepositoryLink
            orgId="org-slug"
            provider={
              Object {
                "config": Array [
                  Object {
                    "help": "Enter your repository name, including the owner.",
                    "label": "Repository Name",
                    "name": "name",
                    "placeholder": "e.g. getsentry/sentry",
                    "required": true,
                    "type": "text",
                  },
                ],
                "id": "github",
                "name": "GitHub",
              }
            }
          />
        </MenuItem>
      </DropdownLink>
    }
    label="Repositories"
  />
  <div
    className="m-b-2"
  >
    <p>
      Connecting a repository allows Sentry to capture commit data via webhooks. This enables features like suggested assignees and resolving issues via commit message. Once you've connected a repository, you can associate commits with releases via the API.
      Â 
      <span>
        <span>
          See our 
        </span>
        <a
          href="https://docs.sentry.io/learn/releases/"
        >
          <span>
            documentation
          </span>
        </a>
        <span>
           for more details.
        </span>
      </span>
    </p>
  </div>
  <div
    className="well blankslate align-center p-x-2 p-y-1"
  >
    <div
      className="icon icon-lg icon-git-commit"
    />
    <h3>
      Sentry is better with commit data
    </h3>
    <p>
      Adding one or more repositories will enable enhanced releases and the ability to resolve Sentry Issues via git message.
    </p>
    <p
      className="m-b-1"
    >
      <a
        className="btn btn-default"
        href="https://docs.sentry.io/learn/releases/"
      >
        Learn more
      </a>
    </p>
  </div>
</div>
`;

exports[`OrganizationRepositories renders with github provider 1`] = `
<div>
  <SettingsPageHeading
    action={
      <DropdownLink
        alwaysRenderMenu={true}
        anchorRight={true}
        caret={true}
        className="btn btn-primary btn-sm"
        disabled={false}
        title="Add Repository"
      >
        <MenuItem
          noAnchor={true}
        >
          <AddRepositoryLink
            orgId="org-slug"
            provider={
              Object {
                "config": Array [
                  Object {
                    "help": "Enter your repository name, including the owner.",
                    "label": "Repository Name",
                    "name": "name",
                    "placeholder": "e.g. getsentry/sentry",
                    "required": true,
                    "type": "text",
                  },
                ],
                "id": "github",
                "name": "GitHub",
              }
            }
          />
        </MenuItem>
      </DropdownLink>
    }
    label="Repositories"
  />
  <Styled(div)>
    <PanelHeader
      disablePadding={true}
    >
      <Flex>
        <Box
          px={2}
        >
          Added Repositories
        </Box>
      </Flex>
    </PanelHeader>
    <Styled(div)>
      <Box />
    </Styled(div)>
  </Styled(div)>
</div>
`;

exports[`OrganizationRepositories renders without providers 1`] = `
<div>
  <SettingsPageHeading
    action={
      <DropdownLink
        alwaysRenderMenu={true}
        anchorRight={true}
        caret={true}
        className="btn btn-primary btn-sm"
        disabled={false}
        title="Add Repository"
      />
    }
    label="Repositories"
  />
  <Styled(div)>
    <PanelHeader
      disablePadding={true}
    >
      <Flex>
        <Box
          px={2}
        >
          Added Repositories
        </Box>
      </Flex>
    </PanelHeader>
    <Styled(div)>
      <Box />
    </Styled(div)>
  </Styled(div)>
</div>
`;
